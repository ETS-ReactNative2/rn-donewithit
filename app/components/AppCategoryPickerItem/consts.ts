const CONSTS = {
  ICON_NAME: 'email',
  TEXT_INPUT_FONT_SIZE: 18,
  ICON_SIZE: 38,
  ICON_WRAP_SIZE: 80
}

export default CONSTS
