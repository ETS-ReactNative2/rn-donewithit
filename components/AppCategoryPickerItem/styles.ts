import COLORS from '../../config/colors'
import CONSTS from './consts'
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginVertical: 10
  },
  iconWrap: {
    width: CONSTS.ICON_WRAP_SIZE,
    height: CONSTS.ICON_WRAP_SIZE,
    borderRadius: CONSTS.ICON_WRAP_SIZE / 2,
    backgroundColor: COLORS.PRIMARY,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    marginTop: 6,
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE,
    width: CONSTS.ICON_WRAP_SIZE,
    textAlign: 'center'
  },
  selected: {
    color: COLORS.PRIMARY
  }
})

export default styles
