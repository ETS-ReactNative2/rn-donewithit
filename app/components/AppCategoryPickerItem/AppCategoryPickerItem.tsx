import React, { ReactDOM, ReactNode, useCallback } from 'react'
import {
  GestureResponderEvent,
  TouchableOpacity,
  View,
  StyleSheet
} from 'react-native'
import { AppPickerItemProps } from '../../components/AppPickerItem'
import AppText from '../../components/AppText'
import AppIcon from '../AppIcon'
import styles from './styles'
import COLORS from '../../config/colors'
import CONSTS from './consts'
function AppCategoryPickerItem({
  label,
  selected,
  onPress,
  data
}: AppPickerItemProps): JSX.Element {
  const {
    iconName = CONSTS.ICON_NAME,
    iconColor = COLORS.WHITE,
    iconSize = CONSTS.ICON_SIZE,
    iconBackgroundColor = COLORS.PRIMARY
  } = data
  /**
    "iconBackgroundColor": "lightslategrey",
    "iconName": "application-outline",
    "label": "Other",
    "value": 9,
   */
  const customStyles = StyleSheet.create({
    iconWrap: {
      backgroundColor: iconBackgroundColor
    }
  })
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity onPress={handlePress} style={styles.container}>
      <View style={[styles.iconWrap, customStyles.iconWrap]}>
        <AppIcon name={iconName} size={iconSize} color={iconColor} />
      </View>
      <AppText style={[styles.text, selected && styles.selected]}>
        {label}
      </AppText>
    </TouchableOpacity>
  )
}

export default AppCategoryPickerItem
