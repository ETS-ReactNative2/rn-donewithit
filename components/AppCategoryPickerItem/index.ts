import AppCategoryPickerItem from './AppCategoryPickerItem'

export default AppCategoryPickerItem
