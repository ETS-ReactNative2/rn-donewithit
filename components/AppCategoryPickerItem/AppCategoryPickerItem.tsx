import React, { ReactNode, useCallback } from 'react'
import { GestureResponderEvent, TouchableOpacity } from 'react-native'
import { AppPickerItemProps } from '../../components/AppPickerItem'
import AppText from '../../components/AppText'
import styles from './styles'
function AppCategoryPickerItem({
  label,
  selected,
  onPress
}: AppPickerItemProps): ReactNode {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity onPress={handlePress}>
      <AppText style={[styles.text, selected && styles.selected]}>
        {label}
      </AppText>
    </TouchableOpacity>
  )
}

export default AppCategoryPickerItem
