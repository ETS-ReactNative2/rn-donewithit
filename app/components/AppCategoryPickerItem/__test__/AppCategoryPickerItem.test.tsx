import React from 'react'
import renderer from 'react-test-renderer'
import AppCategoryPickerItem from '../index'
import COLORS from '../../../config/colors'
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon')

const props = {
  selected: true,
  label: 'Furniture',
  data: {
    label: 'Furniture',
    iconName: 'lamp',
    iconBackgroundColor: COLORS.PRIMARY,
    value: 1
  }
}

describe('AppCategoryPickerItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppCategoryPickerItem {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
