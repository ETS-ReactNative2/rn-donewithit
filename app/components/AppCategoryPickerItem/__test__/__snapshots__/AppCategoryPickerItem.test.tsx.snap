// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCategoryPickerItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "tomato",
          "borderRadius": 40,
          "height": 80,
          "justifyContent": "center",
          "width": 80,
        },
        Object {
          "backgroundColor": "tomato",
        },
      ]
    }
  >
    <Text />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Courier",
          "fontSize": 13,
        },
        Array [
          Object {
            "fontSize": 18,
            "marginTop": 6,
            "textAlign": "center",
            "width": 80,
          },
          Object {
            "color": "tomato",
          },
        ],
      ]
    }
  >
    Furniture
  </Text>
</View>
`;
